.gitlab-ci.yml

```diff
@@ -0,0 +1,78 @@
+  - javac -version
+  - 'wget https://archive.apache.org/dist/maven/maven-3/3.2.5/binaries/apache-maven-3.2.5-bin.zip'
+  - unzip -qq apache-maven-3.2.5-bin.zip
+  - rm apache-maven-3.2.5-bin.zip
+  - apache-maven-3.2.5/bin/mvn --version
+  - ./add_jackson_to_dist.sh
+  - export MVN="$PWD/apache-maven-3.2.5/bin/mvn"
+  - export DEPLOY_SETTINGS="$PWD/deploySettings.xml"
+  # change flink version for PR branches to upload test artifacts
+  - |
+    current_ver=$($MVN help:evaluate -Dexpression=project.version -q -DforceStdout)
+    [[ ! $CI_COMMIT_REF_NAME =~ ^release-.*-splunk.*$ ]] && cd tools && sh change-version.sh $current_ver "${current_ver}-$CI_COMMIT_REF_NAME" && cd ..
+  - export FLINK_VERSION=$($MVN help:evaluate -Dexpression=project.version -q -DforceStdout)
+  - export TARBALL_VERSION="flink-$FLINK_VERSION"
+  # required to minimize connection reset errors in aws when mvn downloads jars
+  - sysctl -w net.ipv4.tcp_keepalive_time=60
+  - sysctl -w net.ipv4.tcp_keepalive_intvl=60
+  - sysctl -w net.ipv4.tcp_keepalive_probes=10
+
+build:artifacts:
+  stage: build 
+  artifacts:
+    paths:
+      - build-target/deploy
+    expire_in: 1 day
+  script:
+    # build binary, source, javadoc, and tests jars in order have them available for tar and upload
+    - su -c "$MVN -B -DskipTests -Dfindbugs.skip -Dcheckstyle.skip=true -Denforcer.skip=true -Drat.skip=true -Drat.ignoreErrors=true -Dhttps.protocols=TLSv1.2 -Dhttp.keepAlive=false -DinstallAtEnd=true install"
+    # create the tarball
+    - mkdir $TARBALL_VERSION
+    - cp -r flink-dist/target/flink-*-bin/flink-*/* $TARBALL_VERSION
+    - mkdir build-target/deploy 
+    - tar -czf build-target/deploy/${TARBALL_VERSION}.tgz $TARBALL_VERSION
+
+upload:local:
+  stage: upload
+  dependencies:
+    - build:artifacts
+  script:
+    # upload artifacts
+    - su -c "$MVN -B -Pdocs-and-source -Dartifactory_user=${SVC_ANTIFRAUD_CI_USER} -Dartifactory_pw=${SVC_ANTIFRAUD_CI_PWD} -Dfindbugs.skip -Dcheckstyle.skip=true -Denforcer.skip=true -Drat.skip=true -Drat.ignoreErrors=true -DskipTests -DaltDeploymentRepository=releases.repo.splunk.com::default::https://repo.splunk.com/artifactory/maven-splunk-local --settings $DEPLOY_SETTINGS -DretryFailedDeploymentCount=10 -DdeployAtEnd=true deploy"
+    # this will upload the release tarball jar. The jar will be uploaded to the artifactId folder, eg
+    #   https://repo.splunk.com/artifactory/maven-splunk-local/org/apache/flink/flink-release/<version>/
+    - su -c "$MVN -B -Durl=https://repo.splunk.com/artifactory/maven-splunk-local -DrepositoryId=releases.repo.splunk.com -DgroupId=org.apache.flink -DartifactId=flink-release -Dversion=${TARBALL_VERSION} -Dpackaging=tar.gz -DgeneratePom=true -Dfile=build-target/deploy/${TARBALL_VERSION}.tgz --settings $DEPLOY_SETTINGS -Dartifactory_user=${SVC_ANTIFRAUD_CI_USER} -Dartifactory_pw=${SVC_ANTIFRAUD_CI_PWD} -DdeployAtEnd=true deploy:deploy-file"
+  only:
+    - /^release-.*-splunk.*$/
+
+upload:test:
```

Comment:
`upload:test` seems very generic. Could we change this to `upload:branch` or something relatable?

Reply:
Yeah I think `upload:test` is not super good but I'm not sure `upload:branch` is much better. Any other suggestions? Maybe `upload:PR`?

Reply:
sure! that works. could you add an explanation that if possible stating that only `one set of artifacts per PR are maintained which would be the latest commit` The user would need to go through the long command to see the flag `-DuniqueVersion=false`  otherwise

Reply:
That's a good point, yeah I'll add that